server/server.go

```diff
@@ -321,7 +324,7 @@ func (s *Server) validateServiceAccountRequest(logger *log.Entry, w http.Respons
 
 		w.WriteHeader(http.StatusForbidden)
 		w.Header().Set("Content-Type", "application/json")
-		if err = json.NewEncoder(w).Encode(&ErrorResponse{
+		if err := json.NewEncoder(w).Encode(&ErrorResponse{
```

Comment:
doesn't this blow up because `err` is already declared in the outer scope?

Reply:
Nope. It shadows it instead.

Reply:
ah ok.  How reasonable.

Reply:
Previously it would modify it instead, which resulted in err being nil and the function returning `nil, nil`. Gee, if only there was a way to have the compiler verify exactly one of two value types was returned like that...